tests/Utils/AstFromValueTest.php

```diff
@@ -88,7 +88,7 @@ public function testConvertsFloatValuesToIntOrFloatASTs() : void
         self::assertEquals(new IntValueNode(['value' => '123']), AST::astFromValue(123.0, Type::float()));
         self::assertEquals(new FloatValueNode(['value' => '123.5']), AST::astFromValue(123.5, Type::float()));
         self::assertEquals(new IntValueNode(['value' => '10000']), AST::astFromValue(1e4, Type::float()));
-        self::assertEquals(new FloatValueNode(['value' => '1e+40']), AST::astFromValue(1e40, Type::float()));
+        self::assertEquals(new FloatValueNode(['value' => '1.0E+40']), AST::astFromValue(1e40, Type::float()));
```

Comment:
What happens with the original format? I mean why do we have this change?

Reply:
`GraphQL\Type\Definition\FloatType::serialize` casts to `float` which destroys any original format, and then it is re-cast back to a string using PHPs default float format which is `1.0E+40`.